### PR TITLE
Added redux foundation and user profile/roles

### PR DESCRIPTION
**Pull request by lesleychard**
- Added foundation for redux/redux-thunk implementation
- Added actions/store for auth0 access token, user profile/roles

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@material-ui/core": "^4.11.0",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.56",
```
Comment (BurtonTechnical) [inline package.json:29]: I know it's not part of this changeset, but curious what's in the lab that's in use here. Pagination maybe?

File: pages/index.tsx [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
+  } = useAuth0();
+  const dispatch = useDispatch();
+  const router = useRouter();
+  const auth0 = useSelector((state: RootState) => state.auth0);
+  const user = useSelector((state: RootState) => state.user);
+
+  const getAccessToken = async () => {
+    try {
+      const accessToken = await getAccessTokenSilently({
+        audience: `https://${process.env.AUTH0_DOMAIN}/api/v2/`,
```
Comment (BurtonTechnical) [inline pages/index.tsx:23]: Question of an ignoramus, but one who has recently run into an adjacent problem: Are you using a preprocessor to turn this into a static resource on compile? I know create-react-app supports REACT_APP-prefixed env variables, but I'm assuming this is something slightly more sophisticated.

**Review by BurtonTechnical — APPROVED** [review]

File: pages/index.tsx (hunk 2 above)
Comment (whymarrh) [reply pages/index.tsx:23]: I believe that the compilation is done via Next.js, though you're right in that these might need a prefix. These might need to be prefixed with `NEXT_PUBLIC_` (per [_Exposing Environment Variables to the Browser_](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser)).

**Comment by whymarrh** [comment]
Nothing here seems untoward. I think this looks good!

Assuming everything here _works_ I do see the properties of a store that are necessary: actions broken into started, success, and failure/error states—that's an important starting point. I would suggest looking into typing the `state` more strictly (I see it's explicitly `any`, currently). If that's possible it might prevent a few typos down the road.

File: package.json (hunk 1 above)
Comment (lesleychard) [reply package.json:29]: It's the `Alert` component

File: pages/index.tsx (hunk 2 above)
Comment (lesleychard) [reply pages/index.tsx:23]: Yeah looking at the resource @whymarrh shared I can probably refactor these into a `.env` file at the very least. Not sure if prefixes are necessary as they are working as-is.

**Review by lesleychard — COMMENTED** [review]
> Nothing here seems untoward. I think this looks good!
> 
> Assuming everything here _works_ I do see the properties of a store that are necessary: actions broken into started, success, and failure/error states—that's an important starting point. I would suggest looking into typing the `state` more strictly (I see it's explicitly `any`, currently). If that's possible it might prevent a few typos down the road.

The root state is explicitly typed (`type RootState`), but the state that is currently `any` has to be that way as `createStore` function requires this within the package. I don't know any way around it unless you have any suggestions.

File: pages/index.tsx (hunk 2 above)
Comment (lesleychard) [reply pages/index.tsx]: Fixed in #8c879ac

**Pull request merged**